version.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/urfave/cli"
+)
+
+// version will be populated by the Makefile, read from
+// VERSION file of the source code.
+var version = ""
+
+// gitCommit will be the hash that the binary was built from
+// and will be populated by the Makefile
+var gitCommit = ""
+
+var versionCommand = cli.Command{
+	Name:        "version",
+	Usage:       "output the runtime version",
+	Description: `The version command outputs the runtime version.`,
+	Action:      printVersion,
+}
+
+func printVersion(context *cli.Context) (err error) {
+	if version == "" {
```

Comment:
On Tue, Aug 23, 2016 at 10:46:25AM -0700, Michael Crosby wrote:
> > +// VERSION file of the source code.
> > +var version = ""
> > +
> > +// gitCommit will be the hash that the binary was built from
> > +// and will be populated by the Makefile
> > +var gitCommit = ""
> > +
> > +var versionCommand = cli.Command{
> > +	Name:        "version",
> > +	Usage:       "output the runtime version",
> > +	Description: `The version command outputs the runtime version.`,
> > +	Action:      printVersion,
> > +}
> > +
> > +func printVersion(context *cli.Context) (err error) {
> > +	if version == "" {
>
> Why don't you just do the same logic with the slice and strings.Join
> so we don't have all these verbose if else

The same logic as what?  This check is for “someone build this
withought ‘-X main.version=…’ like we have in the Makefile”, and I
think we want to keep a check for that case.